### DISCUSSION TITLE
Getting error: Error: __filename is not defined when trying to query the postgres DB

### DISCUSSION THREAD

**Discussion by njiang** [prisma/prisma#22398] [Q&A]
### Question

Hello,

I'm new to Prisma. I've setup prisma to introspect my Postgres DB with the DB connection string. It worked fine.  I then generated the prisma client in my node.js project.  When I'm trying to query the DB with the client I got `Error: __filename is not defined` error. Here is my code snippet.  I'm running it as a cron job of the SST serverless framework:  https://docs.sst.dev/cron-jobs.  Not sure if it's the problem.  But I was able to integrate with aws sqs sdk in the same job and it works fine.  Really appreciate your help!

```js
const prisma = new PrismaClient({
    datasources: {
      db: {
        url: "CONNECTION_STRING",
      },
    },
  });
  const allUsers = await prisma.user.findMany();
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = "CONNECTION_STRING"
}

model user {
  id                                      Int                     @id(map: "PK_cace4a159ff9f2512dd42373760") @default(autoincrement())
  name                                    String                  @db.VarChar
  email                                   String                  @db.VarChar
  created_at                              DateTime                @default(now()) @db.Timestamp(6)
  updated_at                              DateTime                @default(now()) @db.Timestamp(6)
  uuid                                    String                  @db.VarChar
  role                                    String                  @default("user") @db.VarChar
  sellerId                                String?                 @db.VarChar
  isActive                                Boolean                 @default(true)
  accessToken                             String?                 @db.VarChar(256)
  refreshToken                            String?                 @db.VarChar(1024)
  rememberMe                              Boolean                 @default(false)
  sessionToken                            String?                 @db.VarChar(1024)
  sessionAccessKeyId                      String?                 @db.VarChar(256)
  sessionSecretKey                        String?                 @db.VarChar(256)
  listingStatusSubscriptionId             String?                 @db.VarChar(256)
  listingIssuesSubscriptionId             String?                 @db.VarChar(256)
  itemInventoryChangeSubscriptionId       String?                 @db.VarChar(256)
  fulfillmentOrderStatusSubscriptionId    String?                 @db.VarChar(256)
  fbaInventoryAvailabilitySubscriptionId  String?                 @db.VarChar(256)
  fbaOutboundShipmentStatusSubscriptionId String?                 @db.VarChar(256)
  sellerApiConnected                      Boolean                 @default(false)
  feedProcessingFinishedSubscriptionId    String?                 @db.VarChar(256)
  syncJobStatus                           String?                 @db.VarChar
  syncJobId                               String?                 @db.VarChar
  syncJobError                            String?                 @db.VarChar
  recurlyAccountCode                      String?                 @db.VarChar
  boxemSyncJobId                          String?                 @unique(map: "UQ_25e741fdaf8062971fb8d6fbeff") @db.VarChar
  reportProcessingFinishedSubscriptionId  String?                 @db.VarChar(256)
  address                                 address[]
  product                                 product[]
  product_sync_job_info                   product_sync_job_info[]
  shipment                                shipment[]
  shipment_plan                           shipment_plan[]
  user_groups_group                       user_groups_group[]

  @@index([uuid], map: "IDX_a95e949168be7b7ece1a2382fe")
}
```

```ts
// Add any relevant Prisma Client queries here
```

prisma                  : 5.7.0
@prisma/client          : 5.7.0
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v18.12.1
Query Engine (Node-API) : libquery-engine 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at ..\..\node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9 (at ..\..\node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.7.0-41.79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Default Engines Hash    : 79fb5193cf0a8fdbef536e4b4a159cad677ab1b9
Studio                  : 0.495.0

**Comment by ludralph** [comment]
Hi @njiang 👋 

Thank you for raising this question.

This is a known issue when using Prisma with SST serverless framework>

- https://github.com/prisma/prisma/issues/18570

Can you please take a look at that issue and see if any of the workarounds suggested work for you?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
